### PR TITLE
Local embedding model from relative paths

### PR DESCRIPTION
**Pull request by Jeadie**
## 🗣 Description
 - Because [text-embeddings-inference](https://github.com/huggingface/text-embeddings-inference) expects all relevant embedding model files to be in a single directory (since its built for HF models), local embedding models are symlinked into a temporary folder. This gives user flexibility into where and how the relevant files are stored.
 - Bug in symlinking: we cannot symlink a relative path properly. 
 
As apart of this change, we must explicitly parse the `from:` field for each embedding type.

 ### Testing
 
 ```yaml
embeddings:
  - name: openai_default
    from: openai
    params:
      openai_api_key: ${ secrets:SPICE_OPENAI_API_KEY }

  - name: openai_model
    from: openai:text-embedding-ada-002
    params:
      openai_api_key: ${ secrets:SPICE_OPENAI_API_KEY }

  - name: local_hf
    from: huggingface:huggingface.co/sentence-transformers/all-MiniLM-L6-v2

  - name: local_rel
    from: file:models/embed/BAAI/bge-small-en-v1.5/model.safetensors
    files:
      - path: models/embed/BAAI/bge-small-en-v1.5/config.json
      - path: models/embed/BAAI/bge-small-en-v1.5/tokenizer.json
  
  - name: local_abs
    from: file:/Users/jeadie/Github/spiceai/models/embed/BAAI/bge-small-en-v1.5/model.safetensors
    files:
      - path: models/embed/BAAI/bge-small-en-v1.5/config.json
      - path: models/embed/BAAI/bge-small-en-v1.5/tokenizer.json
```

### CODE REVIEW

File: crates/llms/src/embeddings/candle/mod.rs [hunk 1]
```diff
@@ -227,7 +246,16 @@ fn link_files_into_tmp_dir(files: HashMap<String, &Path>) -> Result<PathBuf> {
         .context(FailedToInstantiateEmbeddingModelSnafu)?;
 
     for (name, file) in files {
-        symlink(file, temp_dir.path().join(name))
+        let Ok(abs_path) = path::absolute(file) else {
```
Comment (Jeadie) [inline crates/llms/src/embeddings/candle/mod.rs]: This line is the core of the bug.

**Comment by eadgbear** [comment]
@Jeadie Can you also open a docs ticket to go through what each of these options are? I think we're going to need a bit more explanation about the options, how to get the files needed, etc

**Comment by lukekim** [comment]
@Jeadie ensure this is tested on Windows.

**Comment by Jeadie** [comment]
Docs issue: https://github.com/spiceai/docs/issues/431

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**